### PR TITLE
deploy.sh script for k8s deployments

### PR DESCRIPTION
**Pull request by pontusvision**

### CODE REVIEW

File: deployment-scripts/git-set-creds-github.sh [hunk 1]
```diff
@@ -1,5 +1,8 @@
 #!/usr/bin/env bash
 
+# bail out if we encounter any errors
+set -e
```
Comment (daniel-ac-martin) [inline deployment-scripts/git-set-creds-github.sh]: Ideally we should probably default to using `set -euo pipefail`. As that would:
- Fail early.
- Fail early even when piping
- Fail when trying to use a variable that hasn't been defined.

Comment (nefischer) [reply deployment-scripts/git-set-creds-github.sh]: Good point. Amending

File: deployment-scripts/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+function error {
+    colour='\033[0;31m'
+    standard='\033[0m'
+    echo -e "${colour}ERROR: ${@}${standard}" >&2
+}
+
+
+set -eo pipefail
+# automatically export all environment variables
+set -a
```
Comment (daniel-ac-martin) [inline deployment-scripts/deploy.sh]: I think this should probably be moved down to just before the two `source` statements further down.

File: deployment-scripts/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+CDP_DEPLOYMENT_TEMPLATES_DIR=${ENV_BASE_DIR}/cdp-deployment-templates
+rm -rf ${CDP_DEPLOYMENT_TEMPLATES_DIR}
+cd ${ENV_BASE_DIR}
+git clone --branch ${TAG} --depth 1 https://github.com/UKHomeOffice/cdp-deployment-templates.git
+cd -
+
+source "${CDP_DEPLOYMENT_TEMPLATES_DIR}/vars/common.cfg"
+source "${ENV_BASE_DIR}/conf.cfg"
+
```
Comment (daniel-ac-martin) [inline deployment-scripts/deploy.sh:75]: `set +a` ?

File: deployment-scripts/deploy.sh (hunk 2 above)
Comment (nefischer) [reply deployment-scripts/deploy.sh]: Sure. And if we move those statements after the exports, we can also set -euo pipefail to also catch undefined vars

File: deployment-scripts/deploy.sh (hunk 3 above)
Comment (nefischer) [reply deployment-scripts/deploy.sh:75]: sure

**Review by daniel-ac-martin — APPROVED** [review]

**Pull request merged**